File: samples/iOS/FOOPlugin/FOOPlugin.m
//
//  FOOPlugin.m
//  XTSample
//
//  Created by 崔明辉 on 2017/9/27.
//  Copyright © 2017年 UED Center, YY Inc. All rights reserved.
//

#import "FOOPlugin.h"

@implementation FOOPlugin

- (instancetype)initWithJSContext:(JSContext *)context {
    self = [super init];
    if (self) {
        context[@"FOOPlugin"] = self;
    }
    return self;
}

- (NSString *)sayHello {
    return @"FOO >>> Hello, World!";
}

@end

File: android/app/src/main/java/com/opensource/fooplugin/FOOPlugin.kt
package com.opensource.fooplugin

import com.eclipsesource.v8.V8Object
import com.opensource.xtruntime.XTRContext

/**
 * Created by cuiminghui on 2017/9/27.
 */
class FOOPlugin(xtrContext: XTRContext) {

    init {
        val v8Object = V8Object(xtrContext.runtime)
        v8Object.registerJavaMethod(this, "sayHello", "sayHello", arrayOf())
        xtrContext.runtime.add("FOOPlugin", v8Object)
        v8Object.release()
    }

    fun sayHello(): String {
        return "FOO >>> Hello, World!"
    }

}